### PR TITLE
Create backport pull request via working branch

### PR DESCRIPTION
**Pull request by int128**
## Problem to solve
When the head branch is conflicted, we cannot resolve it from GitHub. It is because the head branch is generally protected.

## How to solve
Create a working branch from the head branch, and then create a backport pull request.

### CODE REVIEW

File: open-backport-pull-request/src/run.ts [hunk 1]
```diff
@@ -38,6 +40,21 @@ export const getHeadBranch = (context: Context): string => {
...
 }
 
+const getHeadSha = async (octokit: Octokit, context: Context): Promise<string> => {
+  if (context.eventName === 'workflow_dispatch') {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+    const branch = context.payload.inputs.headBranch as string
+    const { data: ref } = await octokit.rest.git.getRef({
+      owner: context.repo.owner,
+      repo: context.repo.repo,
+      ref: `heads/${branch}`,
```
Comment (int128) [inline open-backport-pull-request/src/run.ts:50]: https://docs.github.com/en/rest/git/refs#get-a-reference

File: open-backport-pull-request/src/run.ts [hunk 2]
```diff
@@ -55,30 +72,68 @@ const hasDiff = async (params: hasDiffParams): Promise<boolean> => {
...
 }
+
 const openPullRequest = async (params: openPullRequestParams): Promise<string | undefined> => {
+  // Create a working branch so that we can edit it if conflicted.
+  // Generally, the head branch is protected and cannot be edited.
+  const workingBranch = `backport-${params.headBranch.replaceAll('/', '-')}-${Date.now()}`
+  await params.octokit.rest.git.createRef({
+    owner: params.context.repo.owner,
+    repo: params.context.repo.repo,
+    ref: `refs/heads/${workingBranch}`,
```
Comment (int128) [inline open-backport-pull-request/src/run.ts:88]: >The name of the fully qualified reference (ie: refs/heads/master).
>https://docs.github.com/en/rest/git/refs#create-a-reference

File: open-backport-pull-request/src/run.ts [hunk 3]
```diff
@@ -55,34 +72,57 @@ const hasDiff = async (params: hasDiffParams): Promise<boolean> => {
...
+
+  core.info(`Creating a pull request ${workingBranch} -> ${params.baseBranch}`)
+  const pr = await params.octokit.rest.pulls.create({
+    owner: params.context.repo.owner,
+    repo: params.context.repo.repo,
+    head: workingBranch,
+    base: params.baseBranch,
+    title: `Backport ${params.headBranch} into ${params.baseBranch}`,
+    body: `Created from https://github.com/${context.repo.owner}/${context.repo.repo}/actions/runs/${context.runId}`,
+  })
```
Comment (int128) [inline open-backport-pull-request/src/run.ts:100]: We don't need to check `UnprocessableEntityError` here, because the working branch name is unique.

File: open-backport-pull-request/src/run.ts [hunk 4]
```diff
@@ -55,34 +72,57 @@ const hasDiff = async (params: hasDiffParams): Promise<boolean> => {
...
+  core.info(`Created ${pr.data.html_url}`)
+
+  core.info(`Requesting a review to ${params.context.actor}`)
   try {
-    const pr = await params.octokit.rest.pulls.create({
+    await params.octokit.rest.pulls.requestReviewers({
```
Comment (yuya-takeyama) [inline open-backport-pull-request/src/run.ts:105]: Nice!

File: open-backport-pull-request/src/run.ts [hunk 5]
```diff
@@ -55,34 +72,57 @@ const hasDiff = async (params: hasDiffParams): Promise<boolean> => {
...
     })
-    return pr.data.html_url
-  } catch (err) {
-    console.error(err)
+  } catch (e) {
+    core.info(`could not request a review to ${params.context.actor}: ${String(e)}`)
+  }
 
-    if (isUnprocessableEntityError(err)) {
-      core.warning('Pull Request already exists')
```
Comment (yuya-takeyama) [inline open-backport-pull-request/src/run.ts:77]: @int128 
Why did you remove this error handling?

Comment (yuya-takeyama) [reply open-backport-pull-request/src/run.ts:77]: Ah, the answer was here:
https://github.com/quipper/monorepo-deploy-actions/pull/745/files#r1002880273

**Review by yuya-takeyama — APPROVED** [review]
